.github/workflows/r_nold.yml

```diff
@@ -0,0 +1,44 @@
+# Run R tests with noLD R. Only triggered by a pull request review
+# See discussion at https://github.com/dmlc/xgboost/pull/6378
+
+name: XGBoost-R-noLD
+
+on:
+  pull_request_review_comment:
+    types: [created]
+
+env:
+  R_PACKAGES: c('XML', 'igraph', 'data.table', 'magrittr', 'ggplot2', 'DiagrammeR', 'Ckmeans.1d.dp', 'vcd', 'testthat', 'lintr', 'knitr', 'rmarkdown', 'e1071', 'cplm', 'devtools', 'float', 'titanic')
+
+jobs:
+  test-R-noLD:
+    if: github.event.comment.body == '/gha run r-nold-test' && contains('OWNER,MEMBER,COLLABORATOR', github.event.comment.author_association)
+    timeout-minutes: 120
+    runs-on: ubuntu-latest
+    container: rhub/debian-gcc-devel-nold
+    steps:
+    - name: Install git and system packages
+      shell: bash
+      run: |
+        apt-get update && apt-get install -y git libcurl4-openssl-dev libssl-dev libssh2-1-dev libgit2-dev libxml2-dev
+
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'true'
+
+    - name: Install dependencies
+      shell: bash
+      run: |
+        cat > install_libs.R <<EOT
+          install.packages(${{ env.R_PACKAGES }},
+                           repos = 'http://cloud.r-project.org',
+                           dependencies = c('Depends', 'Imports', 'LinkingTo'))
+        EOT
+        /tmp/R-devel/bin/Rscript install_libs.R
```

Comment:
@hcho3 did you know you can make `shell` any executable? I think instead of dealing with `cat`-ing a file, this could be

```yaml
- name: Install dependencies
- shell: /tmp/R-devel/bin/Rscript {0}
- run: |
    install.packages(${{ env.R_PACKAGES }}, repos = "http://cloud.r-project.org", dependencies = c("Depends", "Imports", "LinkingTo"))
```

It might be a little simpler, if you want.

Reply:
I learned about that in `r-lib/actions` examples: https://github.com/r-lib/actions/blob/9598b8eeb6d88de7d76d580d84443542bbfdffce/examples/pkgdown.yaml#L26

Reply:
@jameslamb `/tmp/R-devel/bin/Rscript {0}` resulted into a [strange error](https://github.com/hcho3/xgboost/runs/1389003461?check_suite_focus=true):
```
Error: Second path fragment must not be a drive or UNC name. (Parameter 'expression')
```

Reply:
oh sad! Maybe you can't use absolute paths there. Don't worry about it then, not that important either way.